stores/AzureRecorderActions.ts

```diff
@@ -16,34 +17,45 @@ export const startRecording = async (router: NextRouter) => {
   clearTimeout(get().recorderTimeout);
   const { apiKeyAzure, apiKeyAzureRegion } = get();
 
-  let lastTextUpdate = "";
+  let textUpdates: string[] = [];
 
-  const updateText = (text: string, persist: boolean) => {
-    // Remove previous text update
-    const textInputValue = get().textInputValue;
-    const cutText = textInputValue.substring(
-      0,
-      textInputValue.length - lastTextUpdate.length
-    );
+  const persistText = (text: string) => {
+    const effectiveInputValue = `${get().textInputValue} `;
     set((state) => ({
-      textInputValue: `${cutText}${text}${persist ? " " : ""}`,
+      textInputValue: effectiveInputValue,
     }));
```

Comment:
My understanding of the "recognizing" events is that each successive event contains the content of the last (zeroed out at the next "recognizED" event), and can be considered its predecessor's replacement.

That led to my removing the substring/stitching operation here, but let me know if I've missed something!

Reply:
Ha, great catch, appears I completely overcomplicated this!